tools/src/main/scala/scala/scalanative/codegen/CodeGen.scala

```diff
@@ -714,7 +714,8 @@ object CodeGen {
 
     def genCall(genBind: () => Unit, call: Op.Call)(
         implicit fresh: Fresh): Unit = call match {
-      case Op.Call(ty, Val.Global(pointee, _), args, Next.None) =>
+      case Op.Call(ty, Val.Global(pointee, _), args, Next.None)
+          if lookup(pointee) == ty =>
```

Comment:
Can we merge these
```
case Op.Call(ty, Val.Global(pointee, _), args, Next.None) =>
case Op.Call(ty, ptr, args, Next.None) =>
```
and these
```
case Op.Call(ty, Val.Global(pointee, _), args, unwind) =>
case Op.Call(ty, ptr, args, unwind) =>
```
?

The only differences seem to be prefixing pointee with either `@` or `%` and optional bitcast

Reply:
Good call, fixed in 41a732f.